test/TestIntegration.java

```diff
@@ -1004,6 +965,181 @@ public void prefetchMeta() throws Exception {
     assertNotNull(region_info);
   }
 
+
+  /** Basic reverse scan test. Modeled after the basic forward scan. */
+  @Test
+  public void basicReverseScan() throws Exception {
+    client.setFlushInterval(FAST_FLUSH);
+    final PutRequest put1 = new PutRequest(table, "brs1", family, "q", "v1");
+    final PutRequest put2 = new PutRequest(table, "brs2", family, "q", "v2");
+    final PutRequest put3 = new PutRequest(table, "brs3", family, "q", "v3");
+    Deferred.group(client.put(put1), client.put(put2),
+            client.put(put3)).join();
+    // Scan the same 3 rows created above twice.
+    for (int i = 0; i < 2; i++) {
+        LOG.info("------------ iteration #" + i);
+        final Scanner scanner = client.newScanner(table);
+        scanner.setStartKey("brs9");
+        scanner.setStopKey("brs0");
+        scanner.setReversed(true);
+        // Callback class to keep scanning recursively.
+        class cb implements Callback<Object, ArrayList<ArrayList<KeyValue>>> {
+            private int n = 4; // Asserts will start at the last added values
+            public Object call(final ArrayList<ArrayList<KeyValue>> rows) {
+                if (rows == null) {
+                    return null;
+                }
+                n--;
+                try {
+                    assertSizeIs(1, rows);
+                    final ArrayList<KeyValue> kvs = rows.get(0);
+                    final KeyValue kv = kvs.get(0);
+                    assertSizeIs(1, kvs);
+                    assertEq("brs" + n, kv.key());
+                    assertEq("q", kv.qualifier());
+                    assertEq("v" + n, kv.value());
+                    return scanner.nextRows(1).addCallback(this);
+                } catch (AssertionError e) {
+                    // Deferred doesn't catch Errors on purpose, so transform any
+                    // assertion failure into an Exception.
+                    throw new RuntimeException("Asynchronous failure", e);
+                }
+            }
+        }
+        try {
+            // Reversed scanner will return the largest key in the first nextRows call
+            scanner.nextRows(1).addCallback(new cb()).join();
+        } finally {
+            scanner.close().join();
+        }
+    }
+  }
+
+  /** Longer reverse scan, checks that qualifier and value are still in lexico order. */
+  @Test
+  public void reverseLongerScan() throws Exception {
```

Comment:
Can we make the function names more descriptive of what we're testing?
In this case, it looks like you're testing that the columns are still in forward order.